modelapps/mattermost_client_pp.go

```diff
@@ -0,0 +1,123 @@
+package modelapps
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+const (
+	HEADER_ETAG_CLIENT        = "If-None-Match"
+	HEADER_AUTH               = "Authorization"
+
+	API_URL_SUFFIX_V1 = "/api/v1"
+	API_URL_SUFFIX    = API_URL_SUFFIX_V1
+	APPS_PLUGIN_NAME = "com.mattermost.apps"
+)
+
+type ClientPP struct {
+	Url        string       // The location of the server, for example  "http://localhost:8065"
+	ApiUrl     string       // The api location of the server, for example "http://localhost:8065/api/v4"
+	HttpClient *http.Client // The http client
+	AuthToken  string
+	AuthType   string
+	HttpHeader map[string]string // Headers to be copied over for each request
+
+	// TrueString is the string value sent to the server for true boolean query parameters.
+	trueString string
+
+	// FalseString is the string value sent to the server for false boolean query parameters.
+	falseString string
+}
+
+func NewAPIClientPP(url string) *ClientPP {
```

Comment:
What does Client PP stands for?

Reply:
PP-ProxyPlugin (I think Shota introduced it first) - it is the golang client driver for the PP REST API.